python_modules/dagster/dagster/_core/definitions/auto_materialize_policy.py

```diff
@@ -226,8 +229,19 @@ def without_rules(self, *rules_to_remove: "AutoMaterializeRule") -> "AutoMateria
 
     @public
     def with_rules(self, *rules_to_add: "AutoMaterializeRule") -> "AutoMaterializePolicy":
-        """Constructs a copy of this policy with the specified rules added."""
-        return self._replace(rules=self.rules.union(set(rules_to_add)))
+        """Constructs a copy of this policy with the specified rules added. If an instance of a
+        provided rule already exists in this policy, it will be replaced.
```

Comment:
I like this change a lot, just a nit on the docstring: "If an instance of a provided rule with the same type exists on this policy, it will be replaced"